src/api/Middleware/flightController.js

```diff
@@ -1,28 +1,82 @@
-import actionTypes from '../Actions/actionTypes';
+import { connectFlightController, disconnectFlightController, sendFlightControl } from '../Actions';
+import { actionTypes } from '../Actions/actionTypes';
+import { rootOwnerKey } from '../keys';
+
 import api from '../api';
 
-let topic;
+let topic = undefined;
+let nSubscribers = 0;
 
-async function connectToFlightControllerTopic() {
-  topic = api.startTopic('flightcontroller', { type: 'connect' });
+async function connectToFlightControllerTopic(store) {
+  topic = api.startTopic(
+    'flightcontroller',
+    {
+      event: 'start_subscription',
+      type: "connect"
+    }
+  );
+}
+
+const unsubscribe = () => {
+  if (!topic) {
+    return;
+  }
+  topic.talk({
+    event: 'stop_subscription'
+  });
+  topic.cancel();
+}
+
+const refresh = () => {
+  if (topic) {
+    topic.talk({
+      event: 'refresh'
+    });
+  } else {
+    // If we do not have a subscription, we need to create a new topic.
+    const tempTopic = api.startTopic('flightcontroller', {
+      event: 'refresh'
+    });
+  }
```

Comment:
```suggestion
async function connectToFlightControllerTopic(store) {
  topic = api.startTopic(
    'flightcontroller',
    {
      type: "connect"
    }
  );
}

const unsubscribe = () => {
  if (!topic) {
    return;
  }
  topic.talk({
    event: 'disconnect'
  });
  topic.cancel();
}
```

Reply:
There is no "stop_subscription" or "refresh" in the `flightcontrollertopic.cpp` :)